presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2802,6 +2802,306 @@ public void testCollectColumnStatisticsOnInsert()
+        assertQueryFails("ANALYZE " + tableName, ".*Table.*does not exist.*");
+
+        createUnpartitionedTableForAnalyzeTest(tableName);
+
+        // Test partition properties on unpartitioned table
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[])", tableName), ".*Only partitioned table can be analyzed with a partition list.*");
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p1']])", tableName), ".*Only partitioned table can be analyzed with a partition list.*");
+
+        // Drop the partitioned test table
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testAnalyzePartitionedTable()
+    {
+        String tableName = "test_analyze_partitioned_table";
+        createPartitionedTableForAnalyzeTest(tableName);
+
+        // No column stats before running analyze
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s)", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, null, null, null, null, null), " +
+                        "('c_bigint', null, null, null, null, null, null), " +
+                        "('c_double', null, null, null, null, null, null), " +
+                        "('c_timestamp', null, null, null, null, null, null), " +
+                        "('c_varchar', null, null, null, null, null, null), " +
+                        "('c_varbinary', null, null, null, null, null, null), " +
+                        "('p_varchar', 24.0E0, 6.0E0, 0.25E0, null, null, null), " +
+                        "(null, null, null, null, 16.0E0, null, null)");
+
+        // No column stats after running an empty analyze
+        assertUpdate(format("ANALYZE %s WITH (partitions = ARRAY[])", tableName));
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s)", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, null, null, null, null, null), " +
+                        "('c_bigint', null, null, null, null, null, null), " +
+                        "('c_double', null, null, null, null, null, null), " +
+                        "('c_timestamp', null, null, null, null, null, null), " +
+                        "('c_varchar', null, null, null, null, null, null), " +
+                        "('c_varbinary', null, null, null, null, null, null), " +
+                        "('p_varchar', 24.0E0, 6.0E0, 0.25E0, null, null, null), " +
+                        "(null, null, null, null, 16.0E0, null, null)");
+
+        // Run analyze on 3 partitions including the null partition
+        assertUpdate(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p1'], ARRAY['p2'], ARRAY[NULL]])", tableName), 12);
+
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s WHERE p_varchar = 'p1')", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, 2.0E0, 0.5E0, null, null, null), " +
```

Comment:
All these `.0E0` or `E0` adornments are not necessary. Please remove them.

(also, the fact they are not necessary means there's a coercion underneath, so they serve as verification of the types returned from SHOW STATS)